src/ContractInteract/EIP20CoGateway.js

```diff
@@ -70,6 +95,186 @@ class EIP20CoGateway {
     this._redeemRawTx = this._redeemRawTx.bind(this);
   }
 
+  /**
+   * Setup for EIP20CoGateway. Deploys EIP20CoGateway.
+   *
+   * @param {Object} web3 Web3 object.
+   * @param {EIP20CoGatewaySetupConfig} config EIP20CoGateway setup configuration.
+   * @param {Object} txOptions Transaction options.
+   *
+   * @returns {Promise<EIP20CoGateway>} Promise containing the EIP20CoGateway instance that
+   *                                    has been set up.
+   */
+  static setup(web3, config, txOptions = {}) {
+    EIP20CoGateway.validateSetupConfig(config);
+    validateConfigKeyExists(config, 'gateway', 'config');
```

Comment:
Why is this separate?

Reply:
- move check to validateSetupConfig
- when using it from EIP20Gateway, provide a dummy address so the check passes (and add a comment explaining it)